x-pack/auditbeat/processors/add_session_metadata/pkg/processdb/entry_leader_test.go

```diff
@@ -0,0 +1,1247 @@
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package processdb
+
+import (
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/beats/v7/x-pack/auditbeat/processors/add_session_metadata/pkg/procfs"
+	"github.com/elastic/beats/v7/x-pack/auditbeat/processors/add_session_metadata/types"
+	// "github.com/elastic/elastic-agent-libs/logp"
+)
+
+type (
+	createDBFn func(procfs.Reader) DB
+	testFn     func(*testing.T)
+)
+
+const (
+	containerdShimPath = "/bin/containerd-shim-runc-v2"
+	containerdPath     = "/bin/containerd"
+	sshdPath           = "/usr/bin/sshd"
+	lsPath             = "/usr/bin/ls"
+	bashPath           = "/usr/bin/bash"
+	grepPath           = "/usr/bin/grep"
+)
+
+// Entry evaluation tests
+//
+// The entry leader isn't an entirely rigorous conceptual framework but that
+// shortcoming is outweighted by the large and immediate value it provides.
+//
+// The idea is to assign two pieces of data to each process, the "entry meta"
+// and "entry leader", the former of which describes how the user or system
+// that was ultimately responsible for executing this process got into to the
+// box (e.g. ssh, ssm, kubectl exec) and the latter of which describes the
+// process associated with the user or system's initial entry into the "box"
+// (be it a container, VM or otherwise).
+//
+// Generally speaking, the first session leader in a process lineage of an
+// interactive session is an entry leader having an entry meta type depending
+// on its lineage. For example, in the following process tree, "bash" is an
+// entry leader with entry meta type "sshd":
+//
+// systemd            (pid 1 sid 1)
+// \___ sshd          (pid 100 sid 100)
+//      \___ bash     (pid 1000 sid 1000)
+//           \___ vim (pid 1001 sid 1000)
+//
+// Further entry meta types exist for ssm, container runtimes, serial consoles
+// and other ways to get into a "box" (be it a container or actual machine).
+// The entry meta type "init" is assigned to system processes created by the
+// init service (e.g. rsyslogd, sshd).
+//
+// As should probably be apparent, the code to assign an entry meta type to a
+// process is essentially a large amount of conditional logic with a ton of
+// edge cases. It's something we "bolt on" to the linux process model, and thus
+// finicky and highly subject to bugs.
+//
+// Thankfully, writing unit tests for entry leader evaluation is rather
+// straightforward as it's basically a pure function that requires no external
+// infrastructure to test (just create a mock process event with your desired
+// fields set and pass it in).
+//
+// These tests should effectively serve as the spec for how we assign entry
+// leaders. When further entry meta types or cases are added, tests should be
+
+func requireProcess(t *testing.T, db DB, pid uint32, processPath string) {
```

Comment:
These helpers should have `t.Helper()` so that the failure is correctly reported.